include/solvers/mpm_base.h

```diff
@@ -207,6 +216,15 @@ class MPMBase : public MPM {
   double damping_factor_{0.};
   //! Locate particles
   bool locate_particles_{true};
+  //! Absorbing Boundary Variables
+  bool absorbing_boundary_{false};
+  std::shared_ptr<mpm::AbsorbingConstraint> absorbing_constraint_;
+  unsigned absorbing_dir_{0};
+  int absorbing_nset_id_{0};
```

Comment:
Maybe it would be good to make this a vector of node set ids so we can have more than 1 absorbing boundary